### PR TITLE
Consf 61 Persisting infobox data to MW

### PR DESCRIPTION
**Pull request by lizlux**
This PR adds a persist method for saving infobox to mediawiki.
- Removed `title` property from `InfoboxData` class
- Created helper method for POSTing XHR
- Fleshed out `persist` method.
- Removed test.js in favor of the demo
- Added `persistOptions` property to `InfoboxTemplateBuilder`

### CODE REVIEW

File: src/adapters/mediawiki.js [hunk 1]
```diff
@@ -1,13 +1,99 @@
...
+/**
+ * Send request to MW API to save infobox article with new data
+ * @param xmlString {string} New value for the infobox xml
+ * @param infoboxTitle {string} Name of the article where the infobox xml will be saved
+ * @param editToken {string} Needed for authenticating request
+ * @return {Promise}
+ */
+function save(xmlString, infoboxTitle, editToken) {
+	return new Promise(function (resolve, reject) {
+		xhrPost('http://lizlux.liz.wikia-dev.com/api.php', {
```
Comment (lizlux) [inline src/adapters/mediawiki.js:37]: This url is hard coded for now so the demo will work end to end

File: src/adapters/mediawiki.js [hunk 2]
```diff
@@ -1,13 +1,99 @@
...
+ * Get an edit token so we can save an article via MW API
+ * @param infoboxTitle {string} Name of the article where the infobox xml will be saved
  * @return {Promise}
  */
-export function persist(data) {
-	return new Promise((resolve) => {
-		resolve();
+function getEditToken(infoboxTitle) {
+	return new Promise(function (resolve, reject) {
+		xhrPost('http://lizlux.liz.wikia-dev.com/api.php', {
```
Comment (lizlux) [inline src/adapters/mediawiki.js:68]: This url is hard coded for now so the demo will work end to end

File: src/adapters/mediawiki.js [hunk 3]
```diff
@@ -1,13 +1,99 @@
...
+		!isString(infoboxTitle)
+	) {
+		throw new TypeError('Infobox title and xml are required for saving to MediaWiki');
+	}
+
+	return getEditToken(infoboxTitle)
+			.then(save.bind(null, xmlString, infoboxTitle));
+}
+
+window.persist = persist;
```
Comment (kenkouot) [inline src/adapters/mediawiki.js]: remove this exposure to the global namespace.

**Comment by kenkouot** [comment]
This PR seems to have grown in responsibility and encompasses changes to moving the title property from `InfoboxData` to the orchestrating class. Currently this is unmergeable as the property move has not been fully handled. Serializers and deserialzers need to be updated **and** the API for deserialization changes (`deserialize()` returns `{data: InfoboxData, theme: InfoboxTheme}`–as it stands, there is no way to set a title on the instance of `InfoboxTemplateBuilder` on deserialize, without refactoring.).

Please remove the property change from this PR.

**Comment by kenkouot** [comment]
Talked with @lizlux for a better design for `title` on the Core class.

`title`, in this case, is really a property specific to persistence to MediaWiki, so we should move that property into the persistence adapter. To achieve this, we'll need to improve the API for the builder to allow developers to pass in `persistOptions` when constructing `new InfoboxTemplateBuilder`. These persist options will get passed to the active persist adapters.

Using TypeScript syntax for exposition:

``` typescript
interface PersistOptions {
  [idx:string]: any
}

function persist(doc: string, options: PersistOptions): Promise;

interface CoreOptions {
  from: string;
  persistOptions: PersistOptions
}

class Core {
  constructor(params: CoreOptions)
}
```

**Comment by lizlux** [comment]
Note that CI is failing b/c tests aren't fully configured yet.

**Comment by kenkouot** [comment]
LGTM

**Pull request merged**